File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetProgramsUseCase.java
package org.eyeseetea.malariacare.domain.usecase;

import org.eyeseetea.malariacare.domain.boundary.repositories.IProgramRepository;
import org.eyeseetea.malariacare.domain.entity.Program;

import java.util.List;

public class GetProgramsUseCase {
    private IProgramRepository programRepository;

    public GetProgramsUseCase(IProgramRepository programRepository){
        this.programRepository = programRepository;
    }

    List<Program> execute() throws Exception {
        return programRepository.getAll();
    }
}

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetProgramsUseCase.kt
package org.eyeseetea.malariacare.domain.usecase

import org.eyeseetea.malariacare.domain.boundary.repositories.IProgramRepository
import org.eyeseetea.malariacare.domain.entity.Program

class GetProgramsUseCase(private val programRepository: IProgramRepository) {

    @Throws(Exception::class)
    fun execute(): List<Program> {
        return programRepository.getAll()
    }
}
